pkg/test/client.go

```diff
@@ -18,67 +18,270 @@
 package test
 
 import (
+	"context"
+	openapi_v2 "github.com/googleapis/gnostic/OpenAPIv2"
 	"github.com/m88i/nexus-operator/pkg/apis/apps/v1alpha1"
 	routev1 "github.com/openshift/api/route/v1"
 	"k8s.io/api/networking/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/version"
 	"k8s.io/client-go/discovery"
 	discfake "k8s.io/client-go/discovery/fake"
 	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
 	clienttesting "k8s.io/client-go/testing"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
 const (
-	openshiftGroupVersion  = "openshift.io/v1"
-	monitoringGroupVersion = "monitoring.coreos.com/v1alpha1"
+	openshiftGroupVersion = "openshift.io/v1"
 )
 
-type FakeDiscBuilder struct {
-	*discfake.FakeDiscovery
+// FakeClientBuilder allows building a FakeClient according to
+// the desired cluster capabilities
+type FakeClientBuilder struct {
+	initObjs  []runtime.Object
+	scheme    *runtime.Scheme
+	resources []*metav1.APIResourceList
 }
 
-// NewFakeClient will create a new fake client with all needed schemas
-func NewFakeClient(initObjs ...runtime.Object) client.Client {
-	return fake.NewFakeClientWithScheme(GetSchema(), initObjs...)
+// NewFakeClientBuilder will create a new fake client that is aware of minimal resource types
+// and stores initObjs for initialization later
+func NewFakeClientBuilder(initObjs ...runtime.Object) *FakeClientBuilder {
+	s := scheme.Scheme
+	s.AddKnownTypes(v1alpha1.SchemeGroupVersion, &v1alpha1.Nexus{}, &v1alpha1.NexusList{})
```

Comment:
I’m on the phone. But take a look at the Kogito Operator, I’ve changed the way we add schemas there without having to add objects manually like this.